src/quick-lint-js/parse.h

```diff
@@ -2515,8 +2515,26 @@ class parser {
     }
 
     if (this->peek().type == token_type::kw_else) {
+      source_code_span else_span = this->peek().span();
       this->skip();
-      parse_and_visit_body();
+
+      switch (this->peek().type) {
+      default:
+        parse_and_visit_body();
+        break;
+
+      case token_type::left_paren:
```

Comment:
Hmm, this is simpler than I thought. I like your idea of special-casing expression statements beginning with `(`.